src/prefect/orion/models/block_schemas.py

```diff
@@ -49,9 +49,18 @@ async def create_block_schema(
         exclude={"block_type", "id", "created", "updated"},
     )
     definitions = definitions or insert_values["fields"].pop("definitions", None)
-    insert_values["checksum"] = Block._calculate_schema_checksum(
-        insert_values["fields"]
+    checksum = Block._calculate_schema_checksum(insert_values["fields"])
+
+    # Check for existing block schema based on calculated checksum
+    existing_block_schema = await read_block_schema_by_checksum(
+        session=session, checksum=checksum
     )
+    # Return existing block schema if it exists. Allows block schema creation to be called multiple
+    # times for the same schema without errors.
+    if existing_block_schema:
```

Comment:
would `override=True` do the same thing here?

Reply:
It would, but the side effect would be that block type ID and capabilities for a block schema could be updated via this function. I'd prefer to have a separate route to update block schema capabilities, and I don't think the block type ID should ever be updated after block schema creation, so I'd refrain from exposing the overwrite capability from the REST API.